src/backend/joanie/payment/models.py

```diff
@@ -244,6 +237,24 @@ def get_document_context(self, language_code=None):
 
         return merge_dict(base_context, localized_context)
 
+    def generate_document(self):
+        """
+        Generate the invoice document in format PDF bytes.
+        """
+        try:
+            invoice_context = self.get_document_context()
+        except ValueError as exception:
+            logger.error(
+                "Cannot get document context to generate invoice.",
+                exc_info=exception,
+            )
+            return None, None
+
+        file_bytes = issuers.generate_document(
+            name=payment_enums.INVOICE, context=invoice_context
+        )
+        return invoice_context, file_bytes
+
```

Comment:
IMO this should not be in the model. It should be a service and all information needed should be injected as function parameter

Reply:
Okay, i will create a method in utils to generate an invoice document.

Reply:
We will use the method : `generate_document(name=name, context=context)` in `joanie.core.issuers` to manage the generation of documents.
The method in the Invoice model has been taken out.